astropy/io/ascii/daophot.py

```diff
@@ -28,126 +101,83 @@ def update_meta(self, lines, meta):
-                    keyword_dict['value'] = (vals[0] if len(vals) > 2 else "")
-                    table_meta['keywords'][m.group('name')] = keyword_dict
-                    if m.group('name') == 'APERTURES':
-                        self.aperture_values = keyword_dict['value']
+
+        #self.lines = self.get_header_lines(lines)
+        Nlines = len(self.lines)
+        if Nlines > 0:
+            #group the header lines according to their line identifiers (#K, #N, #U, #F or just # (spacer line))
+            get_line_id = lambda s: s.split(None,1)[0]   #function that grabs the line identifier
+            #group lines by the line identifier ('#N', '#U', '#F', '#K') and capture line index
+            gid, groups = zip( *groupmore(get_line_id, self.lines, range(Nlines)) )
+            #groups of lines and their indeces
+            grouped_lines, gix = zip( *groups )
+            #dict of line groups keyed by line identifiers
+            grouped_lines_dict = gld = dict( zip(gid, grouped_lines) )
+
+            #Update the table_meta keywords if necessary
+            if '#K' in grouped_lines_dict:
+                keywords = OrderedDict( map(self.extract_keyword_line, grouped_lines_dict['#K']) )
+                table_meta['keywords'] = keywords
+
+            coldef_dict = self.parse_col_defs( grouped_lines_dict )
+
+            # If there wasn't a #U defined (not sure of DAOphot specification), then
+            # replace the empty line with the right number of ## indicators, which matches
+            # the DAOphot "no unit" tag.
+            #if not len(coldef_dict['#U'])
+            #for i, coldef_line in enumerate(coldef_lines):
+                #if not coldef_line:
+                    #coldef_lines[i] = ['##'] * len(self.names)
+
+            line_ids = ('#N', '#U', '#F')
+            for name, unit, fmt in zip(*map( coldef_dict.get, line_ids )):
+                meta['cols'][name] = {  'unit'  : unit,
+                                        'format': fmt  }
+
+            self.meta = meta
+            self.names = coldef_dict['#N']
+
+    def extract_keyword_line(self, line):
+        '''extract info from a header keyword line (#K) '''
+        m = self.re_header_keyword.match(line)
+        if m:
+            vals = m.group('stuff').strip().rsplit(None, 2)
+            keyword_dict = { 'units': vals[-2],
+                            'format': vals[-1],
+                            'value' : (vals[0] if len(vals) > 2 else "") }
+            return m.group('name'), keyword_dict
 
     def get_cols(self, lines):
         """Initialize the header Column objects from the table ``lines`` for a DAOphot
         header.  The DAOphot header is specialized so that we just copy the entire BaseHeader
         get_cols routine and modify as needed.
 
-        Parameters
-        ----------
-        lines : list
-            List of table lines
-
+        :param lines: list of table lines
+        :returns: list of table Columns
```

Comment:
Ok, thanks for the info.  Will revert to the numpydoc format.  It's more readable anyway!